app/services/authentication/providers/apple.rb

```diff
@@ -0,0 +1,78 @@
+module Authentication
+  module Providers
+    # Apple authentication provider, uses omniauth-apple as backend
+    class Apple < Provider
+      SETTINGS_URL = "https://appleid.apple.com/account/manage".freeze
+      TRUSTED_CALLBACK_ORIGIN = "https://appleid.apple.com".freeze
+      CALLBACK_PATH = "/users/auth/apple/callback".freeze
```

Comment:
I don't this this is used anywhere, was there a reason for adding it?

Reply:
Yes, this was added to bypass `protect_from_forgery` only if the request comes from the trusted ORIGIN **and** only on Apple's callback path.

Bypass declared [here](https://github.com/forem/forem/pull/7952/files#diff-aed97266b0fe72994225a9cac3e4230bR4-R11) and the check is [here](https://github.com/forem/forem/pull/7952/files#diff-aed97266b0fe72994225a9cac3e4230bR117-R123).

This allows us to "give Apple the keys to enter this door vs leaving the door open for any origin".